jobsdb/jobsdb.go

```diff
@@ -100,43 +100,149 @@ var getTimeNowFunc = func() time.Time {
+	//    })
+	AppendInTx(tx AppendSafeTx, jobList []*JobT) error
+
+	// AppendWithRetryEach tries to append all the provided jobs to the database and returns the job uuids which failed
+	AppendWithRetryEach(jobList []*JobT) map[uuid.UUID]string
+
+	// AppendWithRetryEachInTx tries to append all the provided jobs to the database and returns the job uuids which failed, using an existing transaction.
+	// Please ensure that you are using an AppendSafeTx, e.g.
+	//    jobsdb.WithAppendSafeTx(func(tx AppendSafeTx) error {
+	//	      jobsdb.AppendWithRetryEachInTx(tx, jobList)
+	//    })
+	AppendWithRetryEachInTx(tx AppendSafeTx, jobList []*JobT) map[uuid.UUID]string
+
+	// WithUpdateSafeTx prepares an update-safe environment and then starts a transaction
+	// that can be used by the provided function. An update-safe transaction shall be used if the provided function
+	// needs to call UpdateJobStatusInTx.
+	WithUpdateSafeTx(func(tx UpdateSafeTx) error) error
+
+	// UpdateJobStatus updates the provided job statuses
 	UpdateJobStatus(statusList []*JobStatusT, customValFilters []string, parameterFilters []ParameterFilterT) error
-	UpdateJobStatusInTxn(txHandler *sql.Tx, statusList []*JobStatusT, customValFilters []string, parameterFilters []ParameterFilterT) error
-	AcquireUpdateJobStatusLocks()
-	ReleaseUpdateJobStatusLocks()
-	GetPileUpCounts(statMap map[string]map[string]int)
 
+	// UpdateJobStatusInTx updates the provided job statuses in an existing transaction.
+	// Please ensure that you are using an UpdateSafeTx, e.g.
+	//    jobsdb.WithUpdateSafeTx(func(tx UpdateSafeTx) error {
+	//	      jobsdb.UpdateJobStatusInTx(tx, statusList, customValFilters, parameterFilters)
+	//    })
+	UpdateJobStatusInTx(txHandler UpdateSafeTx, statusList []*JobStatusT, customValFilters []string, parameterFilters []ParameterFilterT) error
+
+	/* Queries */
+
+	// GetUnprocessed finds unprocessed jobs. Unprocessed are new
+	// jobs whose state hasn't been marked in the database yet
+	GetUnprocessed(params GetQueryParamsT) []*JobT
+
+	// GetProcessed finds jobs in some state, i.e. not unprocessed
+	GetProcessed(params GetQueryParamsT) []*JobT
+
+	// GetToRetry finds jobs in failed state
 	GetToRetry(params GetQueryParamsT) []*JobT
+
+	// GetToRetry finds jobs in waiting state
 	GetWaiting(params GetQueryParamsT) []*JobT
-	GetProcessed(params GetQueryParamsT) []*JobT
-	GetUnprocessed(params GetQueryParamsT) []*JobT
+
+	// GetExecuting finds jobs in executing state
 	GetExecuting(params GetQueryParamsT) []*JobT
-	GetImportingList(params GetQueryParamsT) []*JobT
+
+	// GetImporting finds jobs in importing state
+	GetImporting(params GetQueryParamsT) []*JobT
+
+	// GetPileUpCounts returns statistics (counters) of incomplete jobs
+	// grouped by workspaceId and destination type
+	GetPileUpCounts(statMap map[string]map[string]int)
+
+	/* Admin */
 
 	Status() interface{}
-	GetIdentifier() string
+	IsHealthy() bool
 	DeleteExecuting(params GetQueryParamsT)
```

Comment:
Shouldn't this also come under Queries section?

Reply:
indeed `IsHealthy` performs a `SELECT` to check on the database's health but I don't believe that it qualifies as a valuable query, thus I grouped it under the `Admin` group, along with `Status` & `DeleteExecuting`, which are methods used for administrative purposes and not for actual business.